packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,134 @@
+import '@/type-extensions'
+
+import { ActionType } from 'hardhat/types'
+import { task, types } from 'hardhat/config'
+import { TASK_LZ_VALIDATE_RPCS } from '@/constants'
+import { createLogger, printBoolean } from '@layerzerolabs/io-devtools'
+import { printLogo } from '@layerzerolabs/io-devtools/swag'
+import WebSocket from 'ws'
+
+interface TaskArguments {
+    timeout: number
+}
+
+const RPC_URL_KEY = 'url'
+
+const HTTP_URL = 'http://'
+const HTTPS_URL = 'https://'
+
+const WS_URL = 'ws://'
+const WSS_URL = 'wss://'
+
+const TIMEOUT = 1000 // 1 second
+
+const logger = createLogger()
+
+const validateHttpsRpcUrl = async (rpcUrl: string, timeout: number): Promise<boolean> => {
+    const controller = new AbortController()
+    const timeoutId = setTimeout(() => controller.abort(), timeout)
+
+    try {
+        const response = await fetch(rpcUrl, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ jsonrpc: '2.0', method: 'eth_blockNumber', params: [], id: 1 }),
+            signal: controller.signal,
+        })
+
+        clearTimeout(timeoutId)
+
+        if (!response.ok) {
+            logger.error(`RPC URL responded with status: ${response.status}`)
+            return false
+        }
+
+        const data = await response.json()
+        if (data.result) {
+            return true
+        }
+
+        logger.error(`RPC URL responded with invalid data: ${JSON.stringify(data)}`)
+        return false
+    } catch (error) {
+        logger.error(
+            `Validation failed for RPC URL: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+        )
+        return false
+    }
+}
+
+const validateWebSocketRpcUrl = async (rpcUrl: string, timeout: number): Promise<boolean> => {
```

Comment:
This currently passes _any_ responsive wss instead of only responsive wss **for rpcs**. I will fix this in a follow up commit!